### ISSUE TITLE
slint-lsp format returns exit code 0 even on invalid input file

### ISSUE THREAD

**Issue by ubruhin** [slint-ui/slint#8229]
### Bug Description

I'm using `slint-lsp format` as a format checker on CI (make CI fail if sources are not formatted) which basically works, but I realized now that if the input is invalid (e.g. syntax errors) it does report the errors on stderr but it still exits with code 0. So it's not very useful as a format checker...

```bash
$ echo "foo" > test.slint

$ slint-lsp format test.slint
foo
error: Parse error: expected a top-level item such as a component, a struct, or a global
 --> test.slint:1:1
  |
1 | foo
  | ^

$ echo $?
0
```

Is that behavior intended for some reason? Or would it be ok to make it returning nonzero on any kind of error?

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.10.0
- Platform/OS: Linux

**Comment by ogoffart** [comment]
I don't think the behaviour was done on purpose.

A lot of things haven't been polished in the formatter.
